.github/workflows/unit-tests.yml

```diff
@@ -257,7 +257,7 @@ jobs:
           . ~/virtualenvs/pandas-dev/bin/activate
           python -m pip install --no-cache-dir -U pip wheel setuptools meson[ninja]==1.2.1 meson-python==0.13.1
           python -m pip install numpy --config-settings=setup-args="-Dallow-noblas=true"
-          python -m pip install --no-cache-dir versioneer[toml] cython python-dateutil pytz pytest>=7.3.2 pytest-xdist>=2.2.0 hypothesis>=6.46.1
+          python -m pip install --no-cache-dir versioneer[toml] cython python-dateutil pytz<2024.2 pytest>=7.3.2 pytest-xdist>=2.2.0 hypothesis>=6.46.1
```

Comment:
pytz 2024.2 is causing a failure in ``test_array.py``
(relating to constructing an array from datetime objects with pytz timezones)

I've pinned in testing for now.

I'm wondering if I should pin it as a dep too?
(as far as I can tell, all pytz did was update their IANA tzdb to 2024.2, so maybe this fixes itself when everyone else does too?)